cirq-core/cirq/experiments/z_phase_calibration_test.py

```diff
@@ -0,0 +1,192 @@
+    new_dist = _trace_distance(dm_final, dm_target)
+
+    # Either we reduced the error or the error is small enough.
+    assert new_dist < original_dist or new_dist < 1e-6
+
+
+@pytest.mark.parametrize(['angles', 'error'], _create_tests(n=10, seed=32432432))
+def test_calibrate_z_phases_no_options(angles, error):
+
+    original_gate = cirq.PhasedFSimGate(**{k: v for k, v in zip(_ANGLES, angles)})
+    actual_gate = cirq.PhasedFSimGate(**{k: v + e for k, v, e in zip(_ANGLES, angles, error)})
+
+    sampler = _TestSimulator(original_gate, actual_gate, seed=0)
+    qubits = cirq.q(0, 0), cirq.q(0, 1)
+    calibrated_gate = calibrate_z_phases(
+        sampler,
+        qubits,
+        original_gate,
+        options=None,
+        n_repetitions=10,
+        n_combinations=10,
+        n_circuits=10,
+        cycle_depths=range(3, 10),
+    )[qubits]
+
+    initial_unitary = cirq.unitary(original_gate)
+    final_unitary = cirq.unitary(calibrated_gate)
+    target_unitary = cirq.unitary(actual_gate)
+    maximally_mixed_state = np.eye(4) / 2
+    dm_initial = initial_unitary @ maximally_mixed_state @ initial_unitary.T.conj()
+    dm_final = final_unitary @ maximally_mixed_state @ final_unitary.T.conj()
+    dm_target = target_unitary @ maximally_mixed_state @ target_unitary.T.conj()
+
+    original_dist = _trace_distance(dm_initial, dm_target)
+    new_dist = _trace_distance(dm_final, dm_target)
+
+    # Either we reduced the error or the error is small enough.
+    assert new_dist < original_dist or new_dist < 1e-6
+
+
+@pytest.mark.parametrize(['angles', 'error'], _create_tests(n=10, seed=32432432))
+def test_calibrate_z_phases_workflow_no_options(angles, error):
+
+    original_gate = cirq.PhasedFSimGate(**{k: v for k, v in zip(_ANGLES, angles)})
+    actual_gate = cirq.PhasedFSimGate(**{k: v + e for k, v, e in zip(_ANGLES, angles, error)})
+
+    sampler = _TestSimulator(original_gate, actual_gate, seed=0)
+    qubits = cirq.q(0, 0), cirq.q(0, 1)
+    result, _ = z_phase_calibration_workflow(
+        sampler,
+        qubits,
+        original_gate,
+        options=None,
+        n_repetitions=1,
+        n_combinations=1,
+        n_circuits=1,
+        cycle_depths=(1, 2),
+    )
+
+    for params in result.final_params.values():
+        assert 'zeta' not in params
+        assert 'chi' not in params
+        assert 'gamma' not in params
+
+
+def test_plot_z_phase_calibration_result():
+    df = pd.DataFrame()
+    qs = cirq.q(0, 0), cirq.q(0, 1), cirq.q(0, 2)
+    df.index = [qs[:2], qs[-2:]]
+    df['cycle_depths_0'] = [[1, 2, 3]] * 2
+    df['fidelities_0'] = [[0.9, 0.8, 0.7], [0.6, 0.4, 0.1]]
+    df['layer_fid_std_0'] = [0.1, 0.2]
+    df['fidelities_c'] = [[0.9, 0.92, 0.93], [0.7, 0.77, 0.8]]
+    df['layer_fid_std_c'] = [0.2, 0.3]
+
+    _, axes = plt.subplots(1, 2)
+    plot_z_phase_calibration_result(before_after_df=df, axes=axes)
+
+    np.testing.assert_allclose(axes[0].lines[0].get_xdata().astype(float), [1, 2, 3])
```

Comment:
It seems the arrays passed to plotting have `dtype=object`, otherwise this should work without `astype(float)`.
Is it feasible to convert to numerical arrays in or before the plotting code?

Reply:
the  arrays are lists of floats/ints ... eveny when I explicitly passed np.array(..., dtype=float) I still got the same issue